src/app/features/account/components/password-reset/password-reset.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, NonNullableFormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+
+import { FormAlert, FormAlertType, getFormAlertsFromHttpErrorResponse, INavigationService, PageUrl } from '~shared/core';
+
+import { AccountResetPasswordPayload, IAccountService } from '../../services/account.interface';
+
+interface ResetPasswordForm {
+  newPassword: FormControl<string>;
+  confirmPassword: FormControl<string>;
+}
+
+@Component({
+  selector: 'app-password-reset',
+  templateUrl: './password-reset.component.html',
+  styleUrls: ['./password-reset.component.scss'],
+})
+export class PasswordResetComponent implements OnInit {
+  public resetPasswordForm!: FormGroup<ResetPasswordForm>;
+  public resetPasswordFormAlerts!: FormAlert[];
+
+  private resetToken!: string;
+
+  constructor(
+    private readonly navigationService: INavigationService,
+    private readonly accountService: IAccountService,
+    private readonly formBuilder: NonNullableFormBuilder,
+    private readonly activatedRoute: ActivatedRoute,
+  ) {
+  }
+
+  public ngOnInit(): void {
+    this.resetToken = this.activatedRoute.snapshot.queryParamMap.get('token') || '';
+    this.resetPasswordForm = this.formBuilder.group({
+      newPassword: ['', Validators.required],
+      confirmPassword: ['', Validators.required],
+    });
+  }
+
+  public onSubmit(): void {
+    if (this.resetPasswordForm.disabled) {
+      return;
+    }
+    this.resetPasswordForm.disable();
+
+    const resetPasswordPayload: AccountResetPasswordPayload = {
+      token: this.resetToken,
+      newPassword: this.resetPasswordForm.value.newPassword as string,
+      confirmPassword: this.resetPasswordForm.value.confirmPassword as string,
+    };
+
+    this.accountService.resetPassword(resetPasswordPayload).subscribe({
+      error: (err: Error): void => this.onResetError(err),
+      complete: (): void => this.onResetComplete(),
+    });
+  }
+
+  private onResetComplete(): void {
+    this.resetPasswordFormAlerts = [
+      { message: 'ACCOUNT_FEATURE.RESET_PASSWORD.SUCCESS_DESCRIPTION', type: FormAlertType.Success },
+    ];
+    // Redirect to login page after timeout.
+    const timeout: number = 2000;
+    setTimeout(() => {
```

Comment:
I know that we don't have notification support in OpenSource, but this behaviour with timeout it's sloppy UX. Let's skip the success message and redirect directly.